core/templates/domain/topic/topic-object.model.ts

```diff
@@ -581,43 +581,51 @@ export class Topic {
     });
   }
 
-  // Reassigns all values within this topic to match the existing
+  // Reassigns all values within a copy topic variable to match this
   // topic. This is performed as a deep copy such that none of the
   // internal, bindable objects are changed within this topic.
-  copyFromTopic(otherTopic: Topic): void {
-    this._id = otherTopic.getId();
-    this.setName(otherTopic.getName());
-    this.setAbbreviatedName(otherTopic.getAbbreviatedName());
-    this.setUrlFragment(otherTopic.getUrlFragment());
-    this.setThumbnailFilename(otherTopic.getThumbnailFilename());
-    this.setThumbnailBgColor(otherTopic.getThumbnailBgColor());
-    this.setDescription(otherTopic.getDescription());
-    this.setLanguageCode(otherTopic.getLanguageCode());
-    this.setPracticeTabIsDisplayed(otherTopic.getPracticeTabIsDisplayed());
-    this.setMetaTagContent(otherTopic.getMetaTagContent());
-    this.setPageTitleFragmentForWeb(otherTopic.getPageTitleFragmentForWeb());
-    this.setSkillSummariesForDiagnosticTest(
-      otherTopic.getSkillSummariesForDiagnosticTest());
-    this._version = otherTopic.getVersion();
-    this._nextSubtopicId = otherTopic.getNextSubtopicId();
-    this.clearAdditionalStoryReferences();
-    this.clearCanonicalStoryReferences();
-    this.clearUncategorizedSkills();
-    this.clearSubtopics();
-
-    this._canonicalStoryReferences = otherTopic.getCanonicalStoryReferences();
-    this._additionalStoryReferences =
-        otherTopic.getAdditionalStoryReferences();
+  createCopyFromTopic(): Topic {
+    let id = this.getId();
+    let name = this.getName();
+    let abbreviatedName = this.getAbbreviatedName();
+    let urlFragment = this.getUrlFragment();
+    let thumbnailFilename = this.getThumbnailFilename();
+    let thumbnailBgColor = this.getThumbnailBgColor();
+    let description = this.getDescription();
+    let languageCode = this.getLanguageCode();
+    let practiceTabIsDisplayed = this.getPracticeTabIsDisplayed();
+    let metaTagContent = this.getMetaTagContent();
+    let pageTitleFragmentForWeb = this.getPageTitleFragmentForWeb();
+    let skillSummariesForDiagnosticTest =
+      this.getSkillSummariesForDiagnosticTest();
+    let version = this.getVersion();
+    let nextSubtopicId = this.getNextSubtopicId();
+
+    let topic = new Topic(
```

Comment:
It would be nicer to call this local variable `newTopic`. It will help other developers understand what's going on here more easily.

Reply:
Done